src/recipe.cpp

```diff
@@ -1839,15 +1839,22 @@ void Recipe::recalcOgFg()
       // being updated for no good reason.
       // NOTE: We are we recalculating all of these on load? Shouldn't we be
       // reading these values from the database somehow?
-      set( "og", "og", _og, false );
+      // NOTE: We always calculate this, so lets not cause
+      // a database dirty when loading recipes. This fixes
+      // starting the app and then closing causing a Save Changes.
+      // set( "og", "og", _og, false );
```

Comment:
I'm still digging into the database. But saving the og and fg in the database doesn't make since. Change the yeast, and the fg will change. Change a grain and the OG will change. These two should always be calculated based on other things, that ARE saved in the database. I think we should always calculate these two based on other changes, and not save them in the database. This isssue I'm trying to solve is that when the app starts, it loads a default recipe. When the code calulates the OG/FG for this recipe, and calls set(), the database is dirtied, meaning if I changed nothing and close the app, it asks to save changes. Changes? I didn't do anything.

I'm just getting started with the code, so please take what I say as a question and not religion. I ran the code change, and the recipes still show the right OG and FG. It seems to be calculated all the time.